lib/dpdkdevice.cc

```diff
@@ -69,29 +69,92 @@ const char *DPDKDevice::get_device_driver()
 
 #define RETA_CONF_SIZE     (ETH_RSS_RETA_SIZE_512 / RTE_RETA_GROUP_SIZE)
 
-int DPDKDevice::set_rss_max(int max)
+int DPDKDevice::dpdk_set_rss_max(int max)
 {
-    struct rte_eth_rss_reta_entry64 reta_conf[RETA_CONF_SIZE];
+	struct rte_eth_rss_reta_entry64 reta_conf[RETA_CONF_SIZE];
     struct rte_eth_dev_info dev_info;
 
-    rte_eth_dev_info_get(port_id, &dev_info);
-    uint16_t reta_size = dev_info.reta_size;
-    uint32_t i;
-    int status;
-    /* RETA setting */
-    memset(reta_conf, 0, sizeof(reta_conf));
+    uint16_t reta_size = dpdk_get_rss_reta_size();
+	uint32_t i;
+	int status;
+	/* RETA setting */
+	memset(reta_conf, 0, sizeof(reta_conf));
     for (i = 0; i < reta_size; i++) {
-        reta_conf[i / RTE_RETA_GROUP_SIZE].mask = UINT64_MAX;
+			reta_conf[i / RTE_RETA_GROUP_SIZE].mask = UINT64_MAX;
     }
-    for (i = 0; i < reta_size; i++) {
-        uint32_t reta_id = i / RTE_RETA_GROUP_SIZE;
-        uint32_t reta_pos = i % RTE_RETA_GROUP_SIZE;
-        uint32_t core_id = i % max;
-        reta_conf[reta_id].reta[reta_pos] = core_id;
-    }
-    /* RETA update */
-    status = rte_eth_dev_rss_reta_update(port_id, reta_conf, reta_size);
-    return status;
+	for (i = 0; i < reta_size; i++) {
+			uint32_t reta_id = i / RTE_RETA_GROUP_SIZE;
+			uint32_t reta_pos = i % RTE_RETA_GROUP_SIZE;
+			uint32_t core_id = i % max;
+			reta_conf[reta_id].reta[reta_pos] = core_id;
+	}
+	/* RETA update */
+	status = rte_eth_dev_rss_reta_update(port_id,
+			reta_conf,
+			reta_size);
+	return status;
+}
+
```

Comment:
It would be nice to include some documentation along with each of these new methods